h/api/storage.py

```diff
@@ -209,12 +160,8 @@ def delete_annotation(request, id_):
     :param id_: the annotation ID
     :type id_: str
     """
-    if _postgres_enabled(request):
-        annotation = fetch_annotation(request, id_, _postgres=True)
-        request.db.delete(annotation)
-
-    legacy_annotation = fetch_annotation(request, id_, _postgres=False)
-    legacy_annotation.delete()
+    annotation = fetch_annotation(request, id_)
+    request.db.delete(annotation)
```

Comment:
Created a card for it: https://trello.com/c/vW2ihM1m/353-improve-pg-annotation-delete-code